packages/testcontainers/src/chains/defid_container.ts

```diff
@@ -83,6 +83,13 @@ export abstract class DeFiDContainer extends DockerContainer {
     await this.container.start()
   }
 
+  async restart (additionalStartOptions: string[]): Promise<void> {
+    // NOTE(surangap)
+    // implement this.container?.restart() here.
+    // for some reason i could not work it out. even though the container is restrted, new command is not passed.
+    // it was the old command running again. If we could make this out, args can be directly passed with restart.
```

Comment:
seems restart will remain the inherited filesystem state, perhaps just stop and start and name it as `stopStart` function??

Reply:
yes, it will keep the data.
the requirement is to restart the container(old data intact) with a new CMD. but even `start()` does work with a non-empty `options`. so the same behaviour for `restart()`. the issue is either in `Dockerode` or related to docker APIs
tried a few workarounds without success. needs more debugging.
I like `stopStart `. kinda gives the idea of old data intact.